scc/cert/certificate_test.go

```diff
@@ -0,0 +1,65 @@
+package cert
+
+import (
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/bls"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCertificate_CanBeGraduallyAccumulatedAndVerified(t *testing.T) {
+	requires := require.New(t)
+	keys := make([]bls.PrivateKey, 6)
+	members := make([]scc.Member, 6)
+	for i := range keys {
+		keys[i] = bls.NewPrivateKey()
+		members[i] = newMember(keys[i], 1)
+	}
+	committee, err := scc.NewCommittee(members...)
+	requires.NoError(err)
+
+	stmt := testStatement(1)
+
+	cert := NewCertificate(stmt)
+	for i, key := range keys {
+		sig := Sign(stmt, key)
+		requires.NoError(cert.Add(scc.MemberId(i), sig))
+
+		err := cert.Verify(committee)
+		if i < 4 { // needs more than 2/3 to be valid
+			requires.Error(err)
+		} else {
+			requires.NoError(err)
+		}
+	}
+}
+
+func TestCertificate_VerifyAuthority_AcceptsDifferentAuthority(t *testing.T) {
```

Comment:
should this test not check that 2 out of 3 members of a committee are needed to validate the certificate ?

Reply:
The 2/3 majority is in detail tested by the function above and the unit-tests for the aggregated signature. All this test is covering is that the certificate handles `authorities` and `producers` correctly, and is not mixing them up.